great_expectations/data_context/data_context.py

```diff
@@ -540,6 +544,28 @@ def save_config_variable(self, config_variable_name, value):
         with open(config_variables_filepath, "w") as config_variables_file:
             yaml.dump(config_variables, config_variables_file)
 
+    def delete_datasource(self,datasource_name=None):
+        """Delete data source 
+        Args:
+
+        Returns:
+        """
+        if datasource_name is None:
+            raise ValueError(
+                "Datasource names must be a datasource name"
+            )
+        else:
+            datasource = self.get_datasource(datasource_name)
+            if datasource:
+               #remove key until we have a delete method on project_config
+               del self._project_config["datasources"][datasource_name]
+               self._project_config["datasources"][datasource_name] = None
```

Comment:
Do not re-add key once datasource has been deleted.

Reply:
Note: we will need a test to codify the expected semantics of comments inside of configured datasources

Reply:
Sticking with remove. Resolved,